src/UI/Components/Diagram/shapes.ts

```diff
@@ -0,0 +1,209 @@
+import { dia, shapes, util } from "@clientio/rappid";
+import { ColumnData } from "./interfaces";
+
+export class Table extends shapes.standard.HeaderedRecord {
+  defaults() {
+    return util.defaultsDeep(
+      {
+        type: "app.Table",
+        columns: [],
+        padding: { top: 40, bottom: 10, left: 10, right: 10 },
+        size: { width: 176 },
+        itemMinLabelWidth: 80,
+        itemHeight: 22,
+        itemOffset: 0,
+        itemOverflow: true,
+        attrs: {
+          root: {
+            magnet: false,
+          },
+          body: {
+            stroke: "#FFF",
+            fill: "#FFF",
+            strokeWidth: 1,
+          },
+          header: {
+            fill: "#0066CC",
+            stroke: "#0066CC",
+            strokeWidth: 1,
+          },
+          headerLabel: {
+            fill: "#FFFFFF",
+            fontFamily: "Lekton",
+            textTransform: "uppercase",
+            fontSize: 12,
+            textWrap: {
+              ellipsis: true,
+              height: 30,
+            },
+          },
+          itemBodies_0: {
+            // SVGRect which is an active magnet
+            // Do not use `true` to prevent CSS effects on hover
+            magnet: "item",
+          },
+          group_1: {
+            // let the pointer events propagate to the group_0
+            // which spans over 2 columns
+            pointerEvents: "none",
+          },
+          itemLabels: {
+            fontFamily: "Lekton",
+            fontSize: 10,
+            fill: "#000000",
+            pointerEvents: "none",
+          },
+          itemLabels_1: {
+            fill: "#7F7F7F",
+            fontSize: 10,
+            fontFamily: "Lekton",
+            textAnchor: "end",
+            x: `calc(0.5 * w - 10)`,
+          },
+          itemLabels_keys: {
+            x: `calc(0.5 * w - 30)`,
+          },
+          iconsGroup_1: {
+            // SVGGroup does not accept `x` attribute
+            refX: "50%",
+            refX2: -26,
+          },
+        },
+      },
+      super.defaults
+    );
+  }
+
+  preinitialize(): void {
+    this.markup = [
+      {
+        tagName: "rect",
+        selector: "body",
+      },
+      {
+        tagName: "rect",
+        selector: "header",
+      },
+      {
+        tagName: "rect",
+        selector: "tabColor",
+      },
+      {
+        tagName: "text",
+        selector: "headerLabel",
+      },
+    ];
+  }
+  /* eslint-disable @typescript-eslint/no-explicit-any */
```

Comment:
I don't know what args are being sent and the situation is similar to toJSON() probably we won't use it as it is from demo, but I don't want to delete it as it could be useful later. I mean if you think the best approach is to delete unused demo methods then it will be fine for me as I don't have super strong preference but I kept them as they are right now

Reply:
Okay, no problem ! We can keep them for now, but we can't forget to clean it up afterwards :)